Docs for new orgs and workspaces redesign aka Org Landing Page

ian-at-airbyte opened the pull request:
## What

This PR updates docs.airbyte.com to be consistent with the new sidebar/org landing page changes in Airbyte. This is a very large PR that touches almost every section of the docs site.

## How

- **New "Organizations and workspaces" section**: Created a new section to improve and expand the discussion of what these objects are in Airbyte. Pays particular regard to Enterprise Flex, data residency, and self-managed data planes. Moved most organization and workspace settings discussions to be children of that new section. In limited cases, a topic seemed thematically more relevant to another part of the docs, so I left it alone. Completely removed the "Account Management" section, because it was almost entirely about orgs and workspaces.

- **Updated instructions and screenshots**: Updated all user, organization, and workspace settings tasks and instructions. Content now passes checks with MarkDownLint. Almost all of this info needed some level of rewrite, which is why this PR touches so many files. Where screenshots seemed valuable, I kept them and updated them. Several seemed unimportant on review, so I just removed them.

- **Mermaid diagram visual improvement**: Added some color and spacing configuration to Mermaid diagrams so I could write some supporting flow diagrams. These changes aren't perfect, but they make the diagrams slightly more appealing than before. More work is needed on Mermaid's appearance, it just wasn't the focus of this task. These changes were applied in the main Docusaurus configuration file, so they would apply globally.

- **Consolidation of redundant content**: Many connector migration guides had the same instructions. Instead of updating them all, I've added instructions on how to pin and update connectors to the Upgrading Connectors page. Then, updated connector migration guides to link to those instructions instead of reproducing them. This triggers review from Breaking Change Reviewers.

- General cleanup of random problem areas, misspellings, typos, etc. that I noticed along the way.

- Made a few additions to Vale's controlled vocabulary, to avoid linting those words.

### Not part of this PR

- This PR does not include new Arcades/interactive demos. I'll do this as a separate set of changes.

## Review guide
<!--
1. `x.py`
2. `y.py`
-->

- The crux of the change is the creation of [the new orgs and workspaces section](https://airbyte-docs-9x9r2ysrl-airbyte-growth.vercel.app/platform/next/organizations-workspaces/). This is probably the area that should receive the most scrutiny.
- Breaking change reviewers should ensure the changes to the migration guides are satisfactory.
- It might be helpful to spot check the steps in other pages

## User Impact
<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them. 
-->

- Accurate documentation and screenshots.

- No changes to page URLs. This is a low risk set of changes.

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [x] YES 💚
- [ ] NO ❌

ian-at-airbyte committed:
Create orgs and workspaces section

ian-at-airbyte committed:
Create the new organizaton and workspace section, move relevant topics under it, update content under Organizations.

ian-at-airbyte committed:
Refined the organization explanation a bit, and added somewhat better colors for Mermaid diagrams

ian-at-airbyte committed:
Update workspace subsection

ian-at-airbyte committed:
Add actual connector update steps to the connector update page, then update all connector migration guides to reference these steps instead of writing their own

ian-at-airbyte committed:
Update a variety of other pages that referenced the old Settings UI

ian-at-airbyte committed:
Merge branch 'master' into docs-orgs-and-workspaces

ian-at-airbyte committed:
Fix a technical error with Cloud

ian-at-airbyte committed:
Fix two broken links

ian-at-airbyte committed:
Fix a few more references to the old UI

ian-at-airbyte committed:
Update screenshots

ian-at-airbyte committed:
Remaining platform screenshot updates or deletions

ian-at-airbyte committed:
Updated and deleting remaining screenshots in the integrations folder. Also fixed and linted a PostgresSQL Cloud topic that was behaving strangely.

ian-at-airbyte committed:
Update Embedded screenshots

ian-at-airbyte committed:
Update steps for platform docs that referenced the old "Settings" option in code font.

ian-at-airbyte committed:
Merge branch 'master' into docs-orgs-and-workspaces

ian-at-airbyte committed:
Update docs/integrations/sources/zendesk-chat-migrations.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ian-at-airbyte committed:
Update docs/platform/cloud/managing-airbyte-cloud/manage-data-residency.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ian-at-airbyte committed:
Update docs/platform/managing-airbyte/connector-updates.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ian-at-airbyte committed:
Update docs/platform/contributing-to-airbyte/developing-locally.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ian-at-airbyte committed:
Update docs/platform/managing-airbyte/connector-updates.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ian-at-airbyte committed:
Update docs/platform/contributing-to-airbyte/developing-locally.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

ian-at-airbyte committed:
Update docs/platform/managing-airbyte/connector-updates.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

bgroff reviewed (approved):
Some questions and comments but nothing blocking.

joeykmh commented on docs/platform/access-management/sso-providers/azure-entra-id.md:
```diff
@@ -62,15 +62,13 @@ Create client credentials so Airbyte can talk to your application.
+1. In Airbyte, click **Organization settings** > **General**.
 
-2. Under **Organization**, click **General**.
+    :::info
+    Currently, this portion of the setup requires an Airbyte employee. Contact Support to proceed.
+    :::
```
Is this correct? Don't the following steps describe a self-serve SSO configuration?

joeykmh commented on docs/platform/cloud/managing-airbyte-cloud/manage-airbyte-cloud-notifications.md:
```diff
@@ -4,61 +4,66 @@ products: all
+If you want Airbyte to notify you of source schema changes, enable the following notifications.
 
-To set up email notifications, click **Settings** and navigate to **Workspace** > **Notifications**.
+- `Automatic Connection Updates`
```
I think this should be `Connection updates`, also the capitalization in the table vs. the items in this list is inconsistent. In the UI we capitalize all words, e.g. `Connection Updates Requiring Action`

joeykmh commented on docs/platform/cloud/managing-airbyte-cloud/manage-airbyte-cloud-notifications.md:
```diff
@@ -4,61 +4,66 @@ products: all
+If these are off, even if you turned on schema update notifications in a connection's settings, Airbyte won't send notifications about schema changes.
 
-![](./assets/notifications-email.png)
+To edit this setting, click **Connections** and select the connection you want to receive notifications for. Click the **Settings** tab on the Connection page. In the **Advanced Settings**, toggle **Schema update notifications**.
```
This is named slightly different in the connection advanced settings:

<img width="582" height="75" alt="image" src="https://github.com/user-attachments/assets/1eb75a11-1b5b-46ca-844a-b66de3eeb400" />

joeykmh commented on docs/platform/using-airbyte/workspaces.md:
```diff
@@ -2,63 +2,86 @@
+You can create, manage, and delete workspaces, and control access to them.
 
-## Add users to your workspace
+## Create a new workspace
```
Might be worth noting that you need an Organization Editor or Organization Admin role to do this.

joeykmh reviewed (approved):
Nice! Left some comments, but nothing blocking

ian-at-airbyte committed:
Add reusable block for connector upgrades

ian-at-airbyte committed:
Update connector migration guides to use a reusable block of text instead of a link.

The UX in the app requires the user to click a link to get to the migration page, so we judged it a bad experience to then immediately ask them to go to another page again. This way still allows for a reusable set of text that can be easily updated, but the end user UX is less demanding.

ian-at-airbyte committed:
Fix a typo

ian-at-airbyte committed:
Added some best practices to assigning org vs. workspace roles

ian-at-airbyte committed:
Implement adjustments from reviewers' comments

ian-at-airbyte reviewed (commented):
All review comments have been addressed and/or responded to. Thank you!

ian-at-airbyte commented:
Just need approval from @katmarkham or @tybernstein as code owners and then I think we're good to merge.
